src/components/MessageInput/hooks/useCommandTrigger.ts

```diff
@@ -0,0 +1,70 @@
+import { useChatContext } from '../../../context/ChatContext';
+import { CommandItem } from '../../CommandItem/CommandItem';
+import type { CommandResponse } from 'stream-chat';
+
+import type { CommandTriggerSetting } from '../../ChatAutoComplete';
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+} from '../../../../types/types';
+
+const useCommandTrigger = <
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends DefaultChannelType = DefaultChannelType,
+  Co extends DefaultCommandType = DefaultCommandType
+>(): CommandTriggerSetting<Co> => {
+  const { channel } = useChatContext<At, Ch, Co>();
```

Comment:
is there a reason we're pulling channel from the chat context as opposed to the channel state context? also it'll need all the generics

Reply:
No reason for pulling it from the channel state context. I can add the generics, just didn't think it was necessary because we ultimately only care about the commands that are returned by `getCommands`, which only need the `Co` generic, so leaving the others as the defaults seemed a little bit easier. That said, I'm happy to include all of them, just wasn't 100% sure about our policy here